boards: arm: Add support for the VBLUno52 board

iotmember opened the pull request:
+ The VBLUno52 board
    nRF52832 ARM Cortex-M4F processor
    Bluetooth Low Energy 5.0
    DAPLink interface
    UNO pinout
    4 power

+ The following samples were tested:
    hello_world
    basic/button
    basic/blinky
    bluetooth/peripheral_hr
    bluetooth/beacon

Signed-off-by: Nam Do <robotden@gmail.com>

iotvietmember commented:
@dbkinder @galak  Would you please check my commit? Thanks all!

iotvietmember commented:
@galak @dbkinder I hope that this commit will be checked soon. Thanks

dbkinder commented on boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst:
```diff
@@ -0,0 +1,109 @@
+the slow clock is 32.768 kHz. The frequency of the main clock
+is 64 MHz.
+
+Supported Features
```
Just checking, while the board may support these features, are all these features supported by Zephyr?  (I'm not aware of any FOTA support in Zephyr)

dbkinder commented on boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst:
```diff
@@ -0,0 +1,109 @@
+- Supports hardware flow control features (RTS/CTS).
+- *Energy monitoring for BLE module by current measurement*
+- FOTA (Firmware over the air): Upgrade firmware over BLE interface.
+- Build good applications with:
```
Simililar question.  This looks like a cut/paste of generic board information, but we should focus on what Zephyr supports (and leave our discussion about other OS support such as Arduino, RIOT-OS, etc.)

dbkinder commented on boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst:
```diff
@@ -0,0 +1,109 @@
+Flashing
+========
+
+The VBLUno52 board has on-board DAPLink (CMSIS-DAP) interface for flashing and debugging.
```
has an on-board

dbkinder commented on boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst:
```diff
@@ -0,0 +1,109 @@
+	+ USB HID (DAP): CMSIS-DAP compliant debug channel.
+	+ USB CDC: Virtual COM port for log, trace and terminal emulation.
+- Supports hardware flow control features (RTS/CTS).
+- *Energy monitoring for BLE module by current measurement*
```
Why is this in italic? (and does Zephyr support this?)

dbkinder commented on boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst:
```diff
@@ -0,0 +1,109 @@
+This tutorial uses the blinky application :ref:`blinky-sample`
+
+See the :ref:`getting_started` for general information on setting up
+ your development environment.
```
delete the extra space before "your"

dbkinder commented on boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst:
```diff
@@ -0,0 +1,109 @@
+Testing the VBLUno52 with Zephyr: buttons, LEDs, UART, BLE
+**********************************************************
+
+ Here are some sample applications that you can use to test different
```
Delete the leading space on these next 8 lines

dbkinder reviewed (changes requested):
Some questions about board features supported by Zephyr, and some editing suggestions

iotvietmember commented:
@dbkinder  I have just fixed those problems

galak commented on boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst:
```diff
@@ -0,0 +1,117 @@
++-----------+------------+----------------------+
+| SPI       | on-chip    | spi                  |
++-----------+------------+----------------------+
+Other hardware features are not supported by the Zephyr kernel.
```
there should be a blank line after the table.

dbkinder replied:
yep.  good catch.

galak reviewed (changes requested):
Need a blank line after table on line 63 of rst

iotmember committed:
boards: arm: Add support for the VBLUno52 board
+ The VBLUno52 board
    nRF52832 ARM Cortex-M4F processor
    Bluetooth Low Energy 5.0
    DAPLink interface
    UNO pinout
    4 power

+ The following samples were tested:
    hello_world
    basic/button
    basic/blinky
    bluetooth/peripheral_hr
    bluetooth/beacon

Signed-off-by: Nam Do <robotden@gmail.com>

iotvietmember commented:
@galak @dbkinder I have just fixed the error of rst. Many thanks

dbkinder reviewed (approved):
+1 for docs

galak commented:
merged a version #782
